controller/src/main/java/io/pravega/controller/server/ControllerService.java

```diff
@@ -127,6 +127,8 @@
         Preconditions.checkNotNull(kvtConfig, "kvTableConfig");
         Preconditions.checkArgument(createTimestamp >= 0);
         Preconditions.checkArgument(kvtConfig.getPartitionCount() > 0);
+        Preconditions.checkArgument(kvtConfig.getPrimaryKeyLength() > 0);
+        Preconditions.checkArgument(kvtConfig.getSecondaryKeyLength() > 0);
```

Comment:
Secondary Key length may be 0. That's a valid value (see `KeyValueTableConfiguration.java` in the client. Primary Key must be greater than 0 (so you have it correct).

Reply:
Updated this check to reflect the same.